### DISCUSSION TITLE
Connection with Garmin MARQ

### DISCUSSION THREAD

**Discussion by christianstrauch** [bareboat-necessities/lysmarine_gen#337] [Q&A]
All,
I am wondering if anyone has (or knows how to) set up a connection between OpenCPN/SignalK/any other component and the Garmin MARQ Captain's AutoPilot and BoatData apps. I'm specifically interested in directly connecting the AutoPilot app as it would allow me to control the autopilot from my watch via the lysmarine installation as a gatway.

**Comment by mgrouch** [comment]
Connect your pi to travel router. On travel router create WiFi access point. Also map your pi to fixed IP address on wifi network by MAC address. 

Set up you watch to connect to same WiFi network. 

Tcp NMEA 0183 stream would be on port 10110 from pi. 

So you can set it as data source for your watch (probably).  

What is your autopilot?

**Comment by christianstrauch** [comment]
I'm not sure this is how it works on the Garmin MARQ watches. Here's what Garmin says about how to pair a watch with devices (this one is about the "Boat Data" app as an example) -
```
Viewing Boat Data on a Garmin Watch
You can connect a compatible Garmin® watch to a compatible chartplotter to view data from the chartplotter.

Bring the Garmin watch within range (3 m) of the chartplotter.
From the watch clock screen, select START > Boat Data > START.
NOTE: If you have already connected to a chartplotter, and would like to connect to a different chartplotter, open the Boat Data screen, hold UP, and select Pair new.
On the chartplotter, select Communications > Wireless Devices > Connect IQ™ Apps > Boat Data > Enable > New Connection.
The chartplotter begins searching for and connecting to the wearable device.

Compare the code shown on the chartplotter with the one shown on the watch.
If the codes match, select Yes to complete the pairing process.
After the devices are paired, they connect automatically when they are turned on and within range.
``` 

Maybe this can point in the right direction: [Getting Started with Connect IQ BLE Development](https://forums.garmin.com/developer/connect-iq/w/wiki/19/getting-started-with-connect-iq-ble-development)

The "Auto Pilot" app on the Garmin MARQ Captain allows me to select one of "MFD/GHC50" or "GHC20" to connect to. Selecting either one, the watch asks to "Enable Search for Remote on GHC20" or "Enable Autopilot Control in Connect IQ Apps" ("Connect IQ Apps" refers to a setting on the Garmin onboard equipment). I'm looking to emulate such Garmin equipment.
To answer your question though (albeit probably unrelated) - I'm using a Raymarine ST1000 - that should be relatively easy to control using the software in the Bareboat distribution using NMEA 0183.

**Comment by christianstrauch** [comment]
Some further research suggests the GHC20 is using ANT to pair with an autopilot remote. So I guess the idea might be to use an ANT+ dongle with the Pi, simulate a GHC20 on it and attempt to pair the MARQ to that...?

**Comment by mgrouch** [comment]
GHC20 is NMEA 2000 and there is SignalK Autopilot control plugin for it.  Could be an option.

**Comment by christianstrauch** [comment]
Just to clarify the setup - I don't have any Garmin autopilots to connect to.
I have:
- The Pi 4 based lysmarine installation
- A Garmin MARQ Captain watch
- Various SeaTalk 1/2 compatible components incl Raymarine ST1000 autopilot connected via a converter to NMEA 0183 to tty on the Pi

What I want is for any of the software on the lysmarine installation to act **as if it was** a GHC20 so my watch can connect to it (as an autopilot remote). So there is nothing I can connect a cable to, and I can't tell the watch to connect to any specific IP address on a network. It just broadcasts and searches for a GHC20. My guess is, it broadcasts on ANT(+). 
So my question is whether anyone has an idea how to possibly **emulate** a GHC20 or GHC50 (not connect to one) on lysmarine, potentially using an ANT dongle, [maybe like this one](https://www.amazon.de/-/en/Receiver-Adapter-Compatible-CycleOps-TrainerRoad-ATN-USB-Zwift/dp/B07D4J7KH6/ref=sr_1_24?crid=CEB8HWZ15FWU&keywords=ant%2B&qid=1692199360&sprefix=ant%2B%2Caps%2C391&sr=8-24), that is exposed as /dev/tty* on the Pi. Of course that's only if my assumption that the MARQ Captain connection would be done via ANT is correct.

**Comment by mgrouch** [comment]
Raymarine autopilots are usually seatalk so same SignalK autopilot control can control seatalk autopilots. 

Now integration of the watch to SignalK is harder. I’d think it’s using Bluetooth radio. 

However protocol is probably not very open. 
however I’ve seen homeassistant integrations with Garmin Connect IQ devices. 

Garmin has a source code repository on GitHub. Something can be useful from there. There are examples how to write applications for your watch in language called Monkey C. 

I think you better off asking for more details on Garmin forum. It’s relatively active.
